File: src/ru/guychokalolo/teacher/Teacher.java
package ru.guychokalolo.teacher;

import java.util.Objects;

public class Teacher {
    private String name;
    private String surName;
    private int age;
    private String _class;
    private int stage;

    public Teacher(String name, String surName, int age, String _class, int stage){
        this.name = name;
        this.surName = surName;
        this.age = age;
        this._class = _class;
        this.stage = stage;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (!(o instanceof Teacher)) return false;
        Teacher teacher = (Teacher) o;
        return name.equals(teacher.name) &&
                surName.equals(surName) &&
                age==age && _class.equals(_class)
                && stage==stage;
    }

    @Override
    public int hashCode() {
        int hash = 28;
        int result = 1;
        result = hash * result +((name==null)? 0 : name.hashCode());
        return result;
    }

    @Override
    public String toString() {
        return "Teacher{" +
                "name='" + name + '\'' +
                ", surName='" + surName + '\'' +
                ", age=" + age +
                ", _class='" + _class + '\'' +
                ", stage=" + stage +
                '}';
    }
}

File: src/ru/guychokalolo/kotlin/kotlin/teacher/Teacher.kt
package teacher



class Teacher(val name:String,
              val surName:String,
              val age:Int,
              val stage:Double,
              val _class:String) {

    override fun equals(other: Any?): Boolean {

        if (this === other) return true
        if (other !is Teacher) return false

        other as Teacher

        if (name != other.name) return false
        if (surName != other.surName) return false
        if (age != other.age) return false
        if (stage != other.stage) return false
        if (_class != other._class) return false

        return true
    }

    override fun hashCode(): Int {
        val prime = 31
        var result = 1

        result = prime * result + (name?.hashCode() ?: 0)
        result = prime * result + (age?.hashCode())
        result = prime * result + (surName?.hashCode())
        result = prime * result + (_class?.hashCode())
        result = prime * result + if (stage?.equals(null)) 0 else stage.hashCode()

        return result

    }

    override fun toString(): String {

        return "$name, $surName, $age, $stage, $_class"
    }
}